core/batch_loader.py

```diff
@@ -129,11 +129,26 @@ def load_batch(self, batch_path):
         batch_name = _normalize_batch_name(batch_name)
         try:
             batch = Batch.objects.get(name=batch_name)
-            _logger.info("Batch already loaded: %s" % batch_name)
-            return batch
-        except Batch.DoesNotExist as e:
+            msg = "Batch already loaded: %s" % batch_name
+            _logger.warn(msg)
+            raise BatchLoaderException(msg)
+        except Batch.DoesNotExist:
+            # New batch will be created below
             pass
 
+        if Job.in_progress().filter(info=batch_name).count() > 0:
```

Comment:
This code can never be hit, at least not with the job setup we have so far: if a batch exists in the database, the "Batch already loaded" error occurs, even if the batch is only in the process of loading or being purged. This could be a problem if we ever need to know that a job is running as opposed to just knowing a load cannot happen.

If we're okay with a more general / less accurate error message, then we don't need to check for a job in progress at all, though I think that would be the wrong solution.